core/src/scss/utilities/mixins/modular-scale/_modular-spacing-new.scss

```diff
@@ -0,0 +1,77 @@
+@charset "UTF-8";
+
+///
+/// Modular Spacing New
+/// Mixins for modular scaling, a unitless system for returning font sizes or spacing (padding or margin) values.
+/// Our modular spacing system does not use the Bourbon modular-scale mixin. Instead we use a manual scale defined in variables/core/_modular-spacing.scss for finetuning.
+/// A mixin for adding spacing (margin or padding) between elements or between an
+/// element and the edge of its container. The amount of spacing is returned
+/// using a unitless scaling system with a curated scale.
+/// Please see core/scss/utilities/variables/core/_modular-spacing.scss for
+/// variable maps.
+///
+/// @name modular-spacing-new
+///
+/// @example scss
+///   @include modular-spacing-new('padding', 2 null);
+///   @include modular-spacing-new('margin', 0 -4 1 2);
+///
+/// @param {string} $type  Spacing type, either 'padding' or 'margin'.
+/// @param {list} $steps - List (1 to 4 values): Either integer (from -4 to 10 including 0) or null. Follows the shorthand notation of padding/margin. Use null to skip a side. A value of 0 returns the modular spacing base value for that breakpoint.
+/// @param {map} $grid - Breakpoint device sass map. Defaults to $su-grid-device.
+///
+/// @group mixin
+@mixin modular-spacing-new($type: 'padding', $steps: 0, $grid: $su-grid-device) {
+  @each $key, $breakpoint in $grid {
```

Comment:
Why the change to only the three steps? We do some fine tweaking at various breakpoints for spacing on top of the modular spacing rules for some of our custom stuff. Wouldn't we still want the granularity or is that just causing too many media queries and specificity?

Reply:
Oh, this part is actually unchanged - I believe the current modular-spacing mixin only writes 3 sets of values (since there are only 3 unique values in modular-spacing-base).

Reply:
AFAICT the current modular spacing uses the `$su-grid-media` as the breakpoint loop. It should have a value at each breakpoint.

```
$su-grid-media: (
  'xs': $su-breakpoint-xs,
  'sm': $su-breakpoint-sm,
  'md': $su-breakpoint-md,
  'lg': $su-breakpoint-lg,
  'xl': $su-breakpoint-xl,
  '2xl': $su-breakpoint-2xl
) !default;
```

Reply:
Heh I can answer that one. I added a comparison in the current mixin to compare the spacing value computed for that breakpoint. If it's the same value as the one in the last breakpoint, then it doesn't write out that media query.

Reply:
`    /// Only write media query if $i = 1 or if $current_base is different from $previous_base.
    @if $current-base != $previous-base {
      $key: nth($bp-list, $i);`
Line 40 in the mixin 😃

Reply:
Ah I see. Thanks for clarifying that.